be2-scala/docs/README.md

```diff
@@ -224,22 +224,117 @@ val f: Future[GraphMessage] = (dbActor ? DbActor.Write(channel, message)).map {
+git commit -am"<message>"
+
+## Note: You may also set a title and description for a specific commit using your favorite IDE (obviously doom emacs ^^). Then save and quit
+git add -A
+git commit
+```
+
+<div align="center">
+  <img alt="Example of git commit command" src="images/example-git-commit.png" width="700" />
+</div>
+
+:information_source: Do not create a commit with 2.000 lines of codes! Short commit with meaningful titles/description are way better and way easier to review
+
+You then finally want to pull/push the changes
+
+```bash
+git pull			# safety check to see if no one applies changes to your branch in your absence
+git push			# push the commited changes
+```
+
+Note that if you just created the branch, you'll need to tell git once where to push the changes using the following command instead of `git push`
+
+```bash
+git push -u origin <branchName>
+```
+
+:warning: You cannot directly push changes on `master`! For that, you need to go through the process of creating a pull request (see below)
+
+---
+
+### Merge a branch into another
+
+Navigate (using `git checkout <branchName>`) to the branch you want the merge **to be applied to** (i.e. the branch that we merge another into!). Then perform the following command to merge the content of branch `x` into the current checked out branch
+
+```bash
+git merge <x>
+```
+
+---
+
+### Create a pull request
+
+Creating a pull request (PR) can only be done with on the [GitHub website](https://github.com/dedis/student_21_pop/pulls) by navigating on "Pull requests". Click on the big green button titled "New pull request" and choose the receiving branch (generally `master`) as well as the branch you want to merge into the latter (e.g. `work-be2-scala-raulinn-test`). Click on "Create pull request".
+
+You can then set a title and description to your PR as well as set a label (e.g. `be2-scala`), an assignee (the person responsible for the code), a project (if any) and ask for a specific reviewer using the right side bar. Click on "Create pull request"
+
+## 5.	Debugging Tips
 
 The best way to "intercept" a `GraphMessage` being processed in the graph is to launch the server in debug mode, and then sending an isolated message to the server triggering the bug.
 
 :information_source: [Hoppscotch](https://hoppscotch.io/realtime/) (Realtime => WebSocket => `ws://localhost:8000/`) is a useful tool to achieve this result
 
 
-## Coding Styles
+## 6.	Coding Styles
 
-??? link + lint?
+A simple way to have a coherent style across the codebase is to use the IDE features of "code cleanup". For example, in IntelliJ, click on the `src/main/scala` folder and then on `Code -> Reformat Code`. You can then check "include subdirectories", "optimize imports", and "cleanup code" checkbox options. Be careful to not apply these changes to `src/test` folder as it transforms the scalatest syntax into a mess difficult to understand.
```

Comment:
is there a standard configuration file for the Scala IDEs?
Otherwise different IDEs might format things differently depending on the computer's settings.

If there's such a configuration (such as [EditorConfig](https://editorconfig.org/)), consider committing it.